File: src/main/java/org/project/model/dto/DoctorDTO.java
package org.project.model.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.project.enums.operation.SortDirection;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DoctorDTO {
    private String staffEntityFullName;
    private Long staffEntityDepartmentEntityId;
    private Long minStarRating;
    private String sortFieldName;
    private SortDirection sortDirection;
}

File: src/main/resources/templates/frontend/assets/js/model/dto/DoctorDTO.js
export class DoctorDTO {
    constructor(staffEntityFullName, staffEntityDepartmentEntityId, minStarRating, sortFieldName, sortDirection) {
        this.staffEntityFullName = staffEntityFullName;
        this.staffEntityDepartmentEntityId = staffEntityDepartmentEntityId;
        this.minStarRating = minStarRating;
        this.sortFieldName = sortFieldName;
        this.sortDirection = sortDirection;
    }

    static fromObject(obj) {
        const doctorDTO = new DoctorDTO();
        if ("staffEntityFullName" in obj) {
            doctorDTO.staffEntityFullName = obj.staffEntityFullName;
        }
        if ("staffEntityDepartmentEntityId" in obj) {
            doctorDTO.staffEntityDepartmentEntityId = obj.staffEntityDepartmentEntityId;
        }
        if ("minStarRating" in obj) {
            doctorDTO.minStarRating = obj.minStarRating;
        }
        if ("sortFieldName" in obj) {
            doctorDTO.sortFieldName = obj.sortFieldName;
        }
        if ("sortDirection" in obj) {
            doctorDTO.sortDirection = obj.sortDirection;
        }
    }
}